GeneratorInterface/RivetInterface/interface/ParticleLevelProducer.h

```diff
@@ -37,11 +37,14 @@ class ParticleLevelProducer : public edm::one::EDProducer<edm::one::SharedResour
   }
 
   const edm::EDGetTokenT<edm::HepMCProduct> srcToken_;
+  const edm::ParameterSet pset_;
 
   reco::Particle::Point genVertex_;
 
-  Rivet::RivetAnalysis* rivetAnalysis_;
-  Rivet::AnalysisHandler analysisHandler_;
+  Rivet::RivetAnalysis* rivetAnalysis_ = nullptr;
```

Comment:
It would be better if these were `std::unique_ptr`

Reply:
Hi Chris, thank you for your suggestions but I get
```
/build/mseidel/CMSSW_11_2_0_pre1/src/GeneratorInterface/RivetInterface/plugins/ParticleLevelProducer.cc:124:49: error: no matching function for call to 'Rivet::AnalysisHandler::addAnalysis(std::unique_ptr<Rivet::RivetAnalysis>&)'
     analysisHandler_->addAnalysis(rivetAnalysis_);
                                                 ^
```